File: parent/3731 -- Find Missing Elements/mine2.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        # Time: O(n)
        # Space: O(n)

        boolArr = [0] * 100
        minim, maxim = 100,0
        for num in nums:
            boolArr[num-1] = 1
            minim = min(num, minim)
            maxim = max(num, maxim)
        
        return [x for x in range(minim, maxim+1) if not boolArr[x-1]]

File: parent/3731 -- Find Missing Elements/mine2.go
package main

func findMissingElements(nums []int) []int {
	minim, maxim := 100, 0
	var boolArr [100]int
	var res []int
	for _, x := range nums {
		minim = min(minim, x)
		maxim = max(maxim, x)
		boolArr[x-1] = 1
	}
	for x := minim; x <= maxim; x++ {
		if boolArr[x-1] == 0 {
			res = append(res, x)
		}
	}

	return res
}
